nomad/state/state_store.go

```diff
@@ -192,6 +192,24 @@ func (s *StateStore) UpdateNodeStatus(index uint64, nodeID, status string) error
 		return fmt.Errorf("index update failed: %v", err)
 	}
 
+	// Update the state of the allocations which are in running state to lost
+	if status == structs.NodeStatusDown {
+		allocs, err := s.AllocsByNode(nodeID)
+		if err != nil {
+			return fmt.Errorf("error retrieving any allocations for the node: %v", nodeID)
+		}
+		for _, alloc := range allocs {
+			copyAlloc := new(structs.Allocation)
+			*copyAlloc = *alloc
+			if alloc.ClientStatus == structs.AllocClientStatusRunning {
```

Comment:
`alloc.TerminalStatus` won't work here since it considers both the server and client status.

Updating this to check if the alloc status is Pending or Running.